exp/internal/controllers/machinepool_controller_noderef_test.go

```diff
@@ -175,13 +177,23 @@ func TestMachinePoolGetNodeReference(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:           "valid provider id, valid aws node, with minReadySeconds",
+			providerIDList: []string{"aws://us-east-1/id-node-1"},
+			expected: &getNodeReferencesResult{
+				references: []corev1.ObjectReference{{Name: "node-1"}},
+				available:  0,
```

Comment:
nit: We are already checking if available = 0 in the previous test case. Is it possible to add another test case where available > 0?

Reply:
Yeah let's do that. It should be roughly the same test case, just with minReadySeconds set to 0

It looks like we never validate the available and replica fields today. Let's start doing that (and then also fix the other test cases, they are just green because we don't check available & ready)